### PR TITLE
Queries and Data

### PR DESCRIPTION
**Pull request by AlucardSanin**
Qquery and update animals table.

**Project requirements**

Add a column species of type string to your animals table. Modify your schema.sql file.
Insert the following data:
Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
Remember to add these insertions to your data.sql file.

- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Captura1

- Inside a transaction:

Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Commit the transaction.
Verify that change was made and persists after commit.

Captura2

- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.

After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

Captura3

- Inside a transaction:

Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.

Captura4

- Update all animals' weight to be their weight multiplied by -1.

Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction

Captura5

Write queries to answer the following questions:

- How many animals are there?
Captura8

- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?

Captura6

- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Captura7

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,4 +8,10 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.50;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT* FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, COUNT(escape_attempts) FROM animals GROUP BY neutered ORDER BY COUNT(escape_attempts) DESC LIMIT 1;
+SELECT species MAX(weight_kg), MIN(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN DATE '1990-01-01' AND '2000-01-01' GROUP BY species;
```
Comment (MaganaAsiati) [inline queries.sql]: #### In The queries.sql

1. Kindly make sure you create queries to answer the following questions in transactions. Please note: if you find another way that is presented in the example that also works, it is fine!
   * Inside a transaction:
     * `UPDATE` the animals table by setting the species column to unspecified, then roll back the change (\*\*for this step students are explicitly asked to verify that change was made and rollbacked, so make sure that they included all steps as in the example below \*\*).

     ```sql
        BEGIN; -- start transaction
     
        UPDATE animals
        SET species = 'unspecified'; -- make change
     
        SELECT species from animals; -- verify that change was made
        ROLLBACK;
     
        SELECT species from animals; -- verify that change was undone
     ```
   * Inside a transaction:
     * `UPDATE` the animals table by setting the species column to digimon for all animals that have a name ending in mon.
     * `UPDATE` the animals table by setting the species column to pokemon for all animals that don't have species already set.
     * Commit the transaction.

     ```sql
        BEGIN;
        UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
        UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 
     
        SELECT species from animals; -- verify that change was made
        COMMIT;
        SELECT species from animals; -- verify that change persists after commit
     ```
   * Inside a transaction delete all records in the animals table, then roll back the transaction.

     ```sql
        BEGIN;
     
        DELETE FROM animals WHERE ;
        SELECT COUNT(*) FROM ANIMALS;
     
        ROLLBACK;
        SELECT COUNT(*) FROM ANIMALS;
     
     ```
   * Inside a transaction:
     * `DELETE` all animals born after Jan 1st, 2022
     * Create a savepoint for the transaction
     * `UPDATE` all animals' weight to be their weight multiplied by -1.
     * Rollback to the savepoint
     * `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
     * Commit transaction

     ```sql
         BEGIN;
     
         DELETE FROM animals WHERE  date_of_birth > '2022/01/01' ;
         SAVEPOINT SP1;
     
         UPDATE animals SET weight_kg = weight_kg * -1; 
         ROLLBACK TO SP1;
         UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 
     
         COMMIT;
     ```

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 
## Hi @AlucardSanin , I'm @MaganaAsiati, your reviewer for this PR  :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close-gif-4

## Highlights

- Descriptive README file  :heavy_check_mark:👨🏻‍💻
- Great work using the right GitHub Flow.:heavy_check_mark:
- No Linter errors :heavy_check_mark:

## Required Changes ♻️

- Kindly add reactions as required in the project follow this [link](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/lessons/how_to_update_and_delete.md) to learn more about transaction :+1: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future. :heavy_check_mark:👨🏻‍💻
- [ ] kindly  Add a meaningful description to your pull request in the future which gives more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @AlucardSanin,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Davidosky007 — COMMENTED** [review]
Hi @AlucardSanin 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Proper use of Gitflow 👍 
* Well written PR description 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**